function-sidecar/pkg/dispatcher/grpc/windowing_processing_time_window.go

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package grpc
+
+import (
+	"github.com/projectriff/riff/message-transport/pkg/message"
+	"time"
+)
+
+// Type ptimeWindow is a WindowingStrategy that is based on elapsed wall clock time.
+type ptimeWindow struct {
```

Comment:
Just a nit, but I'm not sure `ptimeWindow` needs to be a `struct`. I _think_ you could define:
```
type ptimeWindow int
```
and still define a method:
```
func (s *ptimeWindow) ShouldClose(...
```

Reply:
I think I tried (elsewhere) and failed. So did that for consistency.

Will revisit